src/System.Management.Automation/security/Authenticode.cs

```diff
@@ -275,11 +278,14 @@ internal static class SignatureHelper
         /// <exception cref="System.IO.FileNotFoundException">
         /// Thrown if the file specified by argument fileName is not found.
         /// </exception>
-        internal static Signature GetSignature(string fileName, byte[] fileContent)
+        internal static Signature GetSignature(
+            string fileName,
+            byte[] fileContent,
+            bool embeddedSignatureOnly = false)
         {
             Signature signature = null;
 
-            if (fileContent == null)
+            if (!embeddedSignatureOnly && fileContent == null)
```

Comment:
I'd prefer simplify logic:
```suggestion
            if (embeddedSignatureOnly)
            {
                // WinVerifyTrust APIs
                return GetSignatureFromWinVerifyTrust(fileName, fileContent);
            }

            if (fileContent == null)
```

Reply:
This would mean calling `GetSignatureFromWinVerifyTrust` in multiple locations. While this approach is indeed more streamlined, I believe that using a negated boolean check is still quite comprehensible.